Share/NCShareExtension.swift

```diff
@@ -137,9 +137,8 @@ class NCShareExtension: UIViewController {
         }
 
         // HUD
-        IHProgressHUD.set(viewForExtension: self.collectionView)
+        IHProgressHUD.set(viewForExtension: self.view)
```

Comment:
This seems to fix the bug, but I don't like this solution because:
If you have multiple files that take a while to upload you want to be able to scroll through the bottom file-list to check if everything's there / how many files are left etc. Showing the HUD over the entire view makes the list non-intractable and it isn't obvious why and thus  feels like a bug.

Reply:
- The fix (patch) in on IHProgressHUD.
- If you use self.collectionView and move the collectionView the HUD follow the scroll and the effect it's horrible
- The touch on Cell works during upload (every button must be disabled), look:

![22-01-25 20-41-44 6340](https://user-images.githubusercontent.com/8616947/151048329-f2aeba27-dbaa-4927-a229-4b2f9e837330.png)

The self.view works perfectly and the number on HUD show at what number is the upload (and is a good info), nobody would scroll during a file upload.

If the bug is solved, merged.

Reply:
> - The fix (patch) in on IHProgressHUD.
> - If you use self.collectionView and move the collectionView the HUD follow the scroll and the effect it's horrible
> - The touch on Cell works during upload (every button must be disabled), look:

@marinofaggiana 
- I don't understand what you mean
- That's not possible because the HUD view covers the collection view and interrupts any user input. If you move the collection view the entire modal VC moves with it. It's the same behaviour when using the entire view
- I assume that's the create folder? Correct, that would need to be disabled, or even better remove the directory cell entirely when uploading

Reply:
Look:

![Simulator Screen Shot - iPhone 13 - 2022-01-26 at 11 38 15](https://user-images.githubusercontent.com/8616947/151148492-63097e10-ef04-4f1d-9d41-f7f080858ff3.png)